### ISSUE TITLE
[TRAINER] Fit method

### ISSUE THREAD

**Issue by art-test-stack** [art-test-stack/vessel_position_forecast#6]
Handle various training methods / to handle different models

Can eval on test set while training and keep best model based on result on testset
The device must be able to be specified
Train on the best hyper parameters based on the hyperparameter search (Trainer.find hyperparameters)
Train on all dataset (concat train+test) for last training (pb: overfitting)
